### PR TITLE
Makes the chat loading smoother

### PR DESCRIPTION
**Pull request by DominikPanic**
Improves the matching proc to be /tg/'s, moves it to the same place (client/New())
Chat loading upon login and init should be much smoother now.

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -169,6 +169,27 @@ GLOBAL_VAR_INIT(external_rsc_url, TRUE)
 
 	var/full_version = "[byond_version].[byond_build ? byond_build : "xxx"]"
 
+	if(CONFIG_GET(flag/log_access))
+		for(var/I in GLOB.clients)
+			if(!I || I == src)
```
Comment (Rohesie) [inline code/modules/client/client_procs.dm]: Could you trace stack null vars in `GLOB.clients`?

Comment (DominikPanic) [reply code/modules/client/client_procs.dm]: They will never be there otherwise may as well add it to all for loops

Comment (DominikPanic) [reply code/modules/client/client_procs.dm]: But yeah it's /tg/code so why not

Comment (Rohesie) [reply code/modules/client/client_procs.dm]: Could just remove the check as well.

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -172,6 +172,7 @@ GLOBAL_VAR_INIT(external_rsc_url, TRUE)
 	if(CONFIG_GET(flag/log_access))
 		for(var/I in GLOB.clients)
 			if(!I || I == src)
+				stack_trace("null or new client same as existing one in GLOB.clients during client /New()")
```
Comment (Rohesie) [inline code/modules/client/client_procs.dm]: "client same as existing" is expected to happen, but null is not.

**Pull request merged**